packages/react/src/ButtonGroup/ButtonGroup.tsx

```diff
@@ -1,37 +1,52 @@
 import styled from 'styled-components'
 import {get} from '../constants'
-import sx from '../sx'
+import {type SxProp} from '../sx'
+import React from 'react'
+import Box from '../Box'
+import {defaultSxProp} from '../utils/defaultSxProp'
 import type {ComponentProps} from '../utils/types'
+import {useProvidedRefOrCreate} from '../hooks'
 
-const ButtonGroup = styled.div`
+export type ButtonGroupProps = React.PropsWithChildren<SxProp & ComponentProps<typeof StyledButtonGroup>>
+
+const StyledButtonGroup = styled.div<SxProp>`
   display: inline-flex;
   vertical-align: middle;
   isolation: isolate;
 
-  && > * {
+  && :is(button, a) {
     margin-inline-end: -1px;
     position: relative;
     border-radius: 0;
 
-    :first-child {
-      border-top-left-radius: ${get('radii.2')};
-      border-bottom-left-radius: ${get('radii.2')};
-    }
-
-    :last-child {
-      border-top-right-radius: ${get('radii.2')};
-      border-bottom-right-radius: ${get('radii.2')};
-    }
-
     :focus,
     :active,
     :hover {
       z-index: 1;
     }
   }
 
-  ${sx};
+  && > :first-child :is(button, a) {
```

Comment:
Same here - I don't think we need `&&` or `:is`, we can just directly select `button` and `a` children of `:first-child`

```suggestion
  > :first-child button,
  > :first-child a {
```